.vscode/settings.json

```
@@ -0,0 +1,6 @@
+{
+    "python.formatting.provider": "yapf",
+    "[python]": {   
+        "editor.formatOnSave": true
+    }
+}
```

```suggestion
   "files.trimFinalNewlines": true,
    "files.trimTrailingWhitespace": true,
    "files.insertFinalNewline": true,
}
```

This would automatically remove all the missing newlines that git does not like :wink:

---

:) This is one of those files that are generated by the app as opposed to by hand. No way to avoid the missing newline except by opening up the text version as opposed ot UI and adding it in

---

Wait, I didn't understand this, we should add these settings, right?
Also, you should use `files.insertFinalNewline` instead of `files.trimFinalNewlines` no?

---

Ah, yes, I see what you mean. I think the big first step is opening the pandora's box of committing this file, after that we can debate what to put in it.

---

The idea is to make is such that all files hace exactly one line at the end. This first trims away all empty lines and then adds just a single one :)